core/app/c/[communitySlug]/forms/[formSlug]/edit/actions.ts

```diff
@@ -0,0 +1,37 @@
+"use server";
+
+import { logger } from "logger";
+
+import { db } from "~/kysely/database";
+import { isUniqueConstraintError } from "~/kysely/errors";
+import { getLoginData } from "~/lib/auth/loginData";
+import { autoRevalidate } from "~/lib/server/cache/autoRevalidate";
+import { defineServerAction } from "~/lib/server/defineServerAction";
+
+export const updateForm = defineServerAction(async function updateForm({ communityId, name }) {
+	const user = await getLoginData();
+
+	if (!user) {
+		return {
+			error: "Not logged in",
+		};
+	}
+	try {
+		const formUpdateQuery = await autoRevalidate(
+			db
+				.updateTable("forms")
+				.set({ name: name })
+				.where("communityId", "=", communityId)
+				.returning("id")
+		);
+		//
+		await formUpdateQuery.execute();
+	} catch (error) {
+		if (isUniqueConstraintError(error)) {
+			const column = error.constraint === "forms_slug_key" ? "slug" : "name";
+			return { error: `A form with this ${column} already exists. Choose a new ${column}` };
```

Comment:
This logic for rendering the error doesn't make sense here, but checking for the unique constraint still makes sense! Can you just return the `"...this name already exists...` error without the templating?